crates/ln-dlc-node/src/node/dlc_channel.rs

```diff
@@ -71,7 +63,7 @@ where
     }
 
     #[autometrics]
-    pub async fn propose_dlc_channel_collaborative_settlement(
+    pub fn propose_dlc_channel_collaborative_settlement(
```

Comment:
I can see the motivation of removing `async` if the whole body does a `spawn_blocking`, but I'm not sure whether it will lead us to better code down the line.

Setting aside potential misuse (the caller simply forgetting about `spawn_blocking`, which can happen), we are preventing potential optimisations.

🤔 I'm wondering whether it would be better to have 2 await points, one with task that does the `offer_subchannel_close`, and one that sends the message.

Perhaps we could schedule a meeting and align better on our vision?

Reply:
> Setting aside potential misuse (the caller simply forgetting about spawn_blocking, which can happen).

I'm more afraid of the trap that we seem to have fallen into: assuming that it's okay to use `async` by default (particularly when some of our key dependencies (`rust-dlc` and `rust-lightning`) are almost exclusively sync.

> I'm wondering whether it would be better to have 2 await points, one with task that does the offer_subchannel_close, and one that sends the message.

This doesn't make sense to me. Here we have a block that is purely sync. So we should either do the work in a dedicated OS thread in `tokio`'s blocking thread pool. I think we agree on that.

How could splitting the work into two trips to the blocking thread pool help? You will certainly end up with an async task that is almost always idle (which is good), but you didn't actually need it in the first place (there is no async IO happening here) and you could have just done the same work in one trip. I've asked a [related question](https://discord.com/channels/500028886025895936/500336333500448798/1113786926059769916) in the `tokio` Discord. Let's see what they say.